.github/workflows/autorelease-rubygem.yml

```
@@ -0,0 +1,96 @@
+# USAGE
+# -----
+#
+# on:
+#   workflow_dispatch: {}
+#   schedule:
+#     - cron: '00 13 * * 2'
+#
+# jobs:
+#   autorelease:
+#     permissions:
+#       contents: write
+#       pull-requests: write
+#     uses: alphagov/govuk-infrastructure/.github/workflows/autorelease-rubygem.yml@main
+
+name: Auto-release Rubygem
+
+on:
+  workflow_call:
+    inputs:
+      gem_name:
+        required: false
+        type: string
+        default: ${{ github.event.repository.name }}
+
+jobs:
+  release:
+    name: Release Gem
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+        show-progress: false
+    - uses: ruby/setup-ruby@v1
+      with:
+        rubygems: latest
+        bundler-cache: true
+    - name: "Determine HEAD of default branch"
+      id: fetch_default_branch_head
+      run: echo "sha=$(git ls-remote origin HEAD | cut -f 1)" >> "$GITHUB_OUTPUT"
+    - if: github.sha == steps.fetch_default_branch_head.outputs.sha
+      name: "Auto-release rubygem"
+      env:
+        GEM_NAME: ${{ inputs.gem_name }}
+        GH_TOKEN: ${{ github.token }}
+      run: |
+        set -euo pipefail
+
+        CURR_VERSION=$(ruby -e "puts eval(File.read('$GEM_NAME.gemspec')).version")
+        NEXT_VERSION=$(echo $CURR_VERSION | cut -d . -f1).$(echo $CURR_VERSION | cut -d . -f2).$(( $(echo $CURR_VERSION | cut -d . -f3) + 1 ))
+        BRANCH="release-$NEXT_VERSION"
+        PR_TITLE="Release v$NEXT_VERSION"
+        PR_DESCRIPTION="This is an automated PR to bump the version number to $NEXT_VERSION. This is a patch-level bump, as the only changes since v$CURR_VERSION are dependency updates made by Dependabot."
+
+        if ! [ -f CHANGELOG.md ]; then
+          echo "CHANGELOG.md does not exist. Exiting."
+          exit 0
+        fi
+
+        if ! grep -qR --include "version.rb" $CURR_VERSION; then
+          echo "Can't find a version.rb containing the string '$CURR_VERSION'. Exiting."
+          exit 0
+        fi
+
+        if git ls-remote --exit-code origin refs/heads/$BRANCH > /dev/null; then
+          echo "Branch 'release-$NEXT_VERSION' already exists on the remote. Exiting."
+          exit 0
+        fi
+
+        if [ "$(git log v$CURR_VERSION..main --no-merges --format=format:%an | sort | uniq)" != "dependabot[bot]" ]; then
+          echo "Gem has no unreleased changes, or those changes include commits from users other than Dependabot. Exiting."
+          exit 0
+        fi
+
+        find . -type f -iname version.rb -exec sed -i "s/$CURR_VERSION/$NEXT_VERSION/g" {} \;
+
+        printf "# v$NEXT_VERSION\n\n* Update dependencies\n\n" | cat - CHANGELOG.md > NEW_CHANGELOG.md
```

I think we don't tend to put the `v` in the headers.

```suggestion
        printf "# $NEXT_VERSION\n\n* Update dependencies\n\n" | cat - CHANGELOG.md > NEW_CHANGELOG.md
```